### PR TITLE
DPE-2758 better messaging when no bucket

### PR DESCRIPTION
**Pull request by paulomach**
Lib sync/port from k8s [PR#332](https://github.com/canonical/mysql-k8s-operator/pull/332)

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -90,6 +90,16 @@ jobs:
             echo Skipping unstable tests
             echo "mark_expression=not unstable" >> "$GITHUB_OUTPUT"
           fi
+      - name: Setup microceph
+        # setup microceph before tests
+        uses: phvalguima/microceph-action@v0.3.1
```
Comment (shayancanonical) [inline .github/workflows/ci.yaml:95]: may be we should add a TODO to use canonical action when it is available?

File: lib/charms/mysql/v0/s3_helpers.py [hunk 2]
```diff
@@ -31,7 +31,12 @@
 
 # Increment this PATCH version before using `charmcraft publish-lib` or reset
 # to 0 if you are raising the major API version
-LIBPATCH = 4
+LIBPATCH = 5
+
+
+# botocore/urllib3 clutter the logs when on debug
+logging.getLogger("botocore").setLevel(logging.ERROR)
+logging.getLogger("urllib3").setLevel(logging.ERROR)
```
Comment (shayancanonical) [inline lib/charms/mysql/v0/s3_helpers.py:39]: :rocket:

**Review by shayancanonical — APPROVED** [review]

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (paulomach) [reply .github/workflows/ci.yaml:95]: Added as a task to @phvalguima under https://warthogs.atlassian.net/browse/DPE-2951

**Review by taurus-forever — DISMISSED** [review]

**Comment by paulomach** [comment]
Blocked by: https://github.com/canonical/data-platform-workflows/issues/111

**Comment by paulomach** [comment]
> Blocked by: [canonical/data-platform-workflows#111](https://github.com/canonical/data-platform-workflows/issues/111)

Unblocked

File: tox.ini [hunk 3]
```diff
@@ -80,6 +80,7 @@ set_env =
     {[testenv]set_env}
     # Workaround for https://github.com/python-poetry/poetry/issues/6958
     POETRY_INSTALLER_PARALLEL = false
+    LIBJUJU_VERSION_SPECIFIER = {env:LIBJUJU_VERSION_SPECIFIER:3.2.2}
```
[the remark this answers is not in the document]
Comment (paulomach) [reply tox.ini:83]: So this was added in PR #333 and removed on #354 

This branch was most probable branched in between the two - so, probably not needed?

[the remark this answers is not in the document]
Comment (paulomach) [reply tox.ini]: iirc, was related to unit-test - anyway, it didn't broke anything

**Review by taurus-forever — APPROVED** [review]
Nice! Thank you!

**Pull request merged**